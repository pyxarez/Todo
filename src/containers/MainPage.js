import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addCategory, getCategories } from '../store/actions/CategoryActions';
import { getProgress } from '../store/actions/ProgressActions';
import { getTasks } from '../store/actions/TaskListActions';
import toggleShowDone from '../store/actions/ShowDoneActions';

import { toNumberKeyValues } from '../utils/helpers';

import Header from '../components/Header';
import ShowDone from '../components/ShowDone';
import Filter from '../components/Filter';
import Progressbar from '../components/Progressbar';
import TextBox from '../components/TextBox';

import TaskListContainer from './TaskList';
import Sidebar from './Sidebar';
import MainPageCategoryContainer from './MainPageCategory';

import './MainPage.res/style.css';

export class MainPageContainer extends Component {
  static propTypes = {
    params: PropTypes.object.isRequired,
    addCategory: PropTypes.func.isRequired,
    getProgress: PropTypes.func.isRequired,
    toggleShowDone: PropTypes.func.isRequired,
    getTasks: PropTypes.func.isRequired,
    getCategories: PropTypes.func.isRequired,
    progress: PropTypes.string.isRequired,
    showDone: PropTypes.bool.isRequired,
  }

  componentDidMount = () => {
    this.props.getProgress();
  }

  render() {
    const {
      addCategory,
      progress,
      showDone,
      toggleShowDone,
      params
    } = this.props;
    const URLParams = {
      ...toNumberKeyValues({
        id: params.id,
      }),
      category: params.category,
      filter: params.filter
    };
    const { id } = URLParams;

    return (
      <div className='my-main-page-component'>
        <Header title={'To-Do List'}>
          <ShowDone
            handleShowDoneClick={toggleShowDone}
            showDone={showDone}/>
          <Filter
            URLParams={URLParams}/>
          <Progressbar progress={progress}/>
        </Header>
        <div className='main'>
          <div className='main__sidebar'>
            <TextBox
              isFocused={true}
              onClick={addCategory}
              placeholder='Enter category title'/>
            <Sidebar
              categoryContainer={MainPageCategoryContainer}
              currentCategoryId={id}/>
          </div>
          <TaskListContainer
            URLParams={URLParams}
            showDone={showDone}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ progress, showDone }) => {
  return {
    progress,
    showDone
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addCategory: bindActionCreators(addCategory, dispatch),
    getProgress: bindActionCreators(getProgress, dispatch),
    toggleShowDone: bindActionCreators(toggleShowDone, dispatch),
    getTasks: bindActionCreators(getTasks, dispatch),
    getCategories: bindActionCreators(getCategories, dispatch),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MainPageContainer);
