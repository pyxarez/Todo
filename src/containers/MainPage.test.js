import React from 'react';
import { shallow } from 'enzyme';

import { MainPageContainer } from './MainPage'; 

const setup = () => {
  const props = {
    params: {
      id: '0',
      category: 'vassa',
      filter: 'allah'
    },
    addCategory: () => {},
    getProgress: jest.fn(),
    toggleShowDone: () => {},
    getTasks: () => {},
    getCategories: () => {},
    progress: '0%',
    showDone: true
  };
  const wrapper = shallow(<MainPageContainer {...props} />);
  return { 
    props,
    wrapper
  }
};

describe('Containers :: MainPageContainer', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getProgress on component mount', () => {
    const { wrapper, props } = setup();
    wrapper.instance().componentDidMount();

    expect(props.getProgress).toBeCalled();
  });
});