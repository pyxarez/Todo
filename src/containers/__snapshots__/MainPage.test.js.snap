exports[`Containers :: MainPageContainer should render self and subcomponents 1`] = `
<div
  className="my-main-page-component">
  <Header
    title="To-Do List">
    <ShowDone
      handleShowDoneClick={[Function]}
      showDone={true} />
    <Filter
      URLParams={
        Object {
          "category": "vassa",
          "filter": "allah",
          "id": 0,
        }
      } />
    <Progressbar
      progress="0%" />
  </Header>
  <div
    className="main">
    <div
      className="main__sidebar">
      <TextBox
        isFocused={true}
        onClick={[Function]}
        placeholder="Enter category title" />
      <Connect(Sidebar)
        categoryContainer={[Function]}
        currentCategoryId={0} />
    </div>
    <Connect(TaskListContainer)
      URLParams={
        Object {
          "category": "vassa",
          "filter": "allah",
          "id": 0,
        }
      }
      showDone={true} />
  </div>
</div>
`;
